pypeit/core/telluric.py

```diff
@@ -135,22 +134,73 @@ def pca_eval(theta,pca_dict):
+
+    Optionally, this method also trims wavelength to be in within ``wave_min``
+    and ``wave_max`` and pads the data (see ``pad_frac``).
+
+    Args:
+        filename (:obj:`str`):
+            Telluric PCA filename
+        wave_min (:obj:`float`):
+            Minimum wavelength at which the grid is desired
+        wave_max (:obj:`float`):
+            Maximum wavelength at which the grid is desired.
+        pad_frac (:obj:`float`):
+            Percentage padding to be added to the grid boundaries if
+            ``wave_min`` or ``wave_max`` are input; ignored otherwise. The
+            resulting grid will extend from ``(1.0 - pad_frac)*wave_min`` to
+            ``(1.0 + pad_frac)*wave_max``.
+
+    Returns:
+        :obj:`dict`: Dictionary containing the telluric PCA components.
+            - wave_grid: Telluric model wavelength grid
+            - dloglam: Wavelength sampling of telluric model
+            - tell_pad_pix: Number of pixels to pad at edges for convolution
+            - ncomp_tell_pca: Number of PCA components
+            - tell_pca: PCA component vectors
+            - bounds_tell_pca: Maximum/minimum coefficient
+            - coefs_tell_pca: Set of model coefficient values (for prior in future)
+            - teltype: Type of telluric model, i.e. 'PCA'
+    """
+    # load_telluric_grid() takes care of path and existance check
+    hdul = data.load_telluric_grid(filename)
+    wave_grid_full = hdul[1].data
+    pca_comp_full = hdul[0].data
+    ncomp = hdul[0].header['NCOMP']
+    bounds = hdul[2].data
+    model_coefs = hdul[3].data
+
+    ind_lower = np.argmin(np.abs(wave_grid_full - (1.0 - pad_frac)*wave_min)) \
+                    if wave_min is not None else 0
+    ind_upper = np.argmin(np.abs(wave_grid_full - (1.0 + pad_frac)*wave_max)) \
+                    if wave_max is not None else nspec_full
+    wave_grid = wave_grid_full[ind_lower:ind_upper]
+    pca_comp_grid = pca_comp_full[:,ind_lower:ind_upper]
+
+    dwave, dloglam, resln_guess, pix_per_sigma = wvutils.get_sampling(wave_grid)
+    tell_pad_pix = int(np.ceil(10.0 * pix_per_sigma))
+
+    return dict(wave_grid=wave_grid, dloglam=dloglam,
+                tell_pad_pix=tell_pad_pix, ncomp_tell_pca=ncomp,
+                tell_pca=pca_comp_grid, bounds_tell_pca=bounds,
+                coefs_tell_pca=model_coefs, teltype='PCA')
+            
 def read_telluric_grid(filename, wave_min=None, wave_max=None, pad_frac=0.10):
     """
-    Reads in the telluric grid from a file.
+    Reads in the telluric grid from a file. This method is no longer the
+    preferred approach; see "read_telluric_pca" for the PCA mode.
 
     Optionally, this method also trims the grid to be in within ``wave_min``
     and ``wave_max`` and pads the data (see ``pad_frac``).
 
-    .. todo::
-        List and describe the contents of the dictionary in the return
-        description.
-
     Args:
         filename (:obj:`str`):
            Telluric grid filename
```

Comment:
Although this is not due to your PR, could you please add the missing `optional` in the docstring for optional variables?

Reply:
Done.